### ISSUE TITLE
"Browse Drugs" Sdfile - Empty Structures without proper molfile

### ISSUE THREAD

**Issue by ljbellis** [chembl/user_interface#27]
The "Browse Drugs" sfile is made up of compounds with a structure and without a structure. Those where the structure is not known, still need a valid, but empty, molfile to make the sdfile work.

This is how the file looks currently:
![bd1](https://cloud.githubusercontent.com/assets/9843928/11893911/d05fb2d4-a56b-11e5-961e-6fb791acf6e3.jpg)

However, it should look like this:

![bd2](https://cloud.githubusercontent.com/assets/9843928/11893918/d998615c-a56b-11e5-8dc7-8c59f7f43f4d.jpg)

Note the extra lines (blank) and the $$$$

**Comment by mnowotka** [comment]
![screen shot 2015-12-18 at 09 59 41](https://cloud.githubusercontent.com/assets/837119/11894291/20ff2c9a-a56e-11e5-8f14-c8a09662315f.png)

**Comment by ljbellis** [comment]
Sorry, ignore the last $$$$ - this is below the whole lot of properties.

**Comment by ljbellis** [comment]
![bd3](https://cloud.githubusercontent.com/assets/9843928/11896859/0542f8a6-a582-11e5-867b-f9d7099a8c92.jpg)

**Comment by ndedman** [comment]
I think this error propagates from the download_helper class.
There's a specific line of code that adds a dummy molecule entry, but that molecule is malformed:
```php
$molfile = ($row['MOLFILE']) ? $row['MOLFILE']->load() : "  0  0  0  0  0  0            999 V2000\nM  END\n";
```
As you can see, the name/comment lines are missing. We could have:
```php
$molfile = ($row['MOLFILE']) ? $row['MOLFILE']->load() : "\nno_structure\n  0  0  0  0  0  0            999 V2000\nM  END\n$$$$\n";
```
which should allow the whole file to parse.

**Comment by ndedman** [comment]
This is now fixed for all SDF exports and returns a valid but empty molfile.

Also, if there is a filter applied to a table, this is now reflected in the export for that table (txt/sdf). However, since the download_helper code is a bit of a mess, I've implemented it in a naive way (regular expression on a concatenated version of the string). Performance wise, it's suboptimal (it could be a lot faster, but the SQL and logic is getting messy without a complete re-write), but it's quite fast.
